src/OpenTelemetry/Resources/OtelEnvResourceDetector.cs

```diff
@@ -38,15 +39,74 @@ private static List<KeyValuePair<string, object>> ParseResourceAttributes(string
         string[] rawAttributes = resourceAttributes.Split(AttributeListSplitter);
         foreach (string rawKeyValuePair in rawAttributes)
         {
-            string[] keyValuePair = rawKeyValuePair.Split(AttributeKeyValueSplitter);
-            if (keyValuePair.Length != 2)
+#if NETSTANDARD2_1 || NET8_0_OR_GREATER
```

Comment:
There is no need to specify .NET version. There is a support only for .NET8+.
What is more, if you want to take it for .NETStandard 2.1, it will be great  to be opened for future version (I doubt, that it will happen, but...)

```suggestion
#if NETSTANDARD2_1_OR_GREATER || NET
```